app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -245,6 +225,57 @@ private static boolean shouldGenerateFirstKey() {
         return keyServer == null || keyServer.equals("");
     }
 
+    private ResultAndError<PullTaskResult> getRequestResultOrRetry(AndroidTransactionParserFactory factory) {
+        if (retryLimit != -1 && numTries >= retryLimit) {
+            // for testing purposes only

```

Comment:
sorry; I'm confused why this is for testing purposes only. It seems like a legit thing to be checking.